### ISSUE TITLE
Remove DotCom My Sites Dropdown when WP.com Toolbar enabled on Jetpack connected sites with Nav Unification

### ISSUE THREAD

**Issue by getdave** [Automattic/wp-calypso#50519]
Please see the very long thread here for context p1HpG7-bcl-p2#comment-44902.

When the `Enable the WordPress.com toolbar` setting is toggled "on" under JP settings on a self-hosted JP connected site then the WP.com toolbar replaces the WPAdmin toolbar. This causes a `My Sites` item to appear (top lef) which when clicked reveals a menu which doesn't match up with the unified nav or the unmodified WPAdmin nav (note the nav is _intentionally_ unmodified in WPAdmin for _non-Atomic_ JP connected sites).

<img width="958" alt="Screen Shot 2021-02-26 at 10 27 13" src="https://user-images.githubusercontent.com/444434/109288770-37c56480-781d-11eb-9092-62f80fd8c36b.png">

The simplest approach to deal with this is to remove the `My Sites` dropdown, which means changing:

From this:

```
DotCom Toolbar
DotCom My Sites Dropdown
WP Admin sidebar menu (untouched)
```

To this:

```
DotCom Toolbar
WP Admin sidebar menu (untouched)
```

Note this is not a change introduced by Nav Unification and thus is not a blocker for launch.

**Comment by cpapazoglou** [comment]
This is part of calypsoify and it is removed while `nav-unification` is active here

https://github.com/Automattic/jetpack/blob/75adba6a8e9bd27d0d53b48c8f9556749d9a78aa/projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php#L735-L740

**Comment by pablinos** [comment]
@cpapazoglou Now Calypsoify is disabled and nav unification is enabled for all, do you think we can close this?

**Comment by cpapazoglou** [comment]
@pablinos while users with nav unifications won't ever see this, JP connected non-Atomic sites still have this behavior. Is that expected or is it a good time to remove it?

**Comment by pablinos** [comment]
Ah, I see. I'm not sure this is Calypsoify related. It might be that we change behaviour based on Calypsoify as I know that we disable that menu on Atomic sites (or at least we did before the nav unification launch).

I think removing it would be a good idea. The My Sites link can simply redirect to Calypso, and that's one less menu to think about!

**Comment by cpapazoglou** [comment]
> I think removing it would be a good idea. The My Sites link can simply redirect to Calypso, and that's one less menu to think about!

Could we replicate the UX of how it works in Simple & Calypso, meaning:
- While in the frontend, it redirects to Calypso
- While in WP Admin, it does nothing (since you are already seeing the unified sidebar)

**Comment by pablinos** [comment]
> While in WP Admin, it does nothing (since you are already seeing the unified sidebar)

Does that depend on whether nav unification is active? Perhaps when it's inactive (on non-Atomic Jetpack sites) it redirects to Calypso?

**Comment by cpapazoglou** [comment]
> > While in WP Admin, it does nothing (since you are already seeing the unified sidebar)
> 
> Does that depend on whether nav unification is active? Perhaps when it's inactive (on non-Atomic Jetpack sites) it redirects to Calypso?

Good point @pablinos, when it's inactive "Browse Sites" is missing. In that case, redirecting to Calypso sounds fine.
